backend/compact-connect/tests/unit/test_api.py

```diff
@@ -54,54 +55,25 @@ def _generate_expected_integration_object(self, handler_logical_id: str) -> dict
             }
         }
 
-    def test_synth_generates_provider_users_resource(self):
-        api_stack = self.app.sandbox_stage.api_stack
-        api_stack_template = Template.from_stack(api_stack)
-
-        # Ensure the resource is created with expected path
-        api_stack_template.has_resource_properties(
-            type=CfnResource.CFN_RESOURCE_TYPE_NAME,
-            props={
-                "ParentId": {
-                    # Verify the parent id matches the expected 'v1' resource
-                    'Ref': api_stack.get_logical_id(api_stack.api.v1_api.resource.node.default_child)
-                },
-                "PathPart": "provider-users"
-            })
-
-    def test_synth_generates_get_provider_users_me_endpoint_resource(self):
-        api_stack = self.app.sandbox_stage.api_stack
-        api_stack_template = Template.from_stack(api_stack)
-
-        # Ensure the resource is created with expected path
-        api_stack_template.has_resource_properties(
-            type=CfnResource.CFN_RESOURCE_TYPE_NAME,
-            props={
-                "ParentId": {
-                    # Verify the parent id matches the expected 'provider-users' resource
-                    'Ref': api_stack.get_logical_id(api_stack.api.v1_api.provider_users_resource.node.default_child)
-                },
-                "PathPart": "me"
-            })
-
-        # ensure the handler is created
-        api_stack_template.has_resource_properties(
-            type=CfnFunction.CFN_RESOURCE_TYPE_NAME,
-            props={
-                "Handler": "handlers.provider_users.get_provider_user_me"
-            })
-
-        # ensure the GET method is configured with the lambda integration and authorizer
-        api_stack_template.has_resource_properties(
-            type=CfnMethod.CFN_RESOURCE_TYPE_NAME,
-            props={
-                "HttpMethod": "GET",
-                # the provider users endpoints uses a separate authorizer from the staff endpoints
-                "AuthorizerId": {
-                    "Ref": api_stack.get_logical_id(api_stack.api.provider_users_authorizer.node.default_child)
-                },
-                # ensure the lambda integration is configured with the expected handler
-                "Integration": self._generate_expected_integration_object(
-                    api_stack.get_logical_id(
-                        api_stack.api.v1_api.provider_users.get_provider_users_me_handler.node.default_child)
-                )})
+    def compare_snapshot(self, actual: dict, snapshot_name: str, overwrite_snapshot: bool = False):
```

Comment:
Oh, this might be your solution to my prior comment re "testing." Let me know if that's right.

Reply:
Also, looks like this function is only needed in the `test_app.py` file, not here?

Reply:
Yes, this test is how we confirm that the compact configuration matches expected values for the test environment. This code is used by both the `test_app` and the child classes of `test_api` [here](https://github.com/csg-org/CompactConnect/pull/243/files#diff-9dfe2171d4a0ef351c005ebda8009b928b8a129478a27d0aee0c341ecf11c1feR93-R95) and [here](https://github.com/csg-org/CompactConnect/pull/243/files#diff-9ab2f250e93c65ce67302cb35ebd8f91210a98ab3335696ac494f740f2f07304R109-R110)

Reply:
It's worth pointing out that @jusdino will be refactoring all of these tests in an upcoming PR to reduce duplicating stack synth steps and make them easier to maintain in general.